### PR TITLE
fix aot autograd for None fw inputs

### PR DESCRIPTION
**Pull request by bdhirsh**
hot fix: Confirmed this fixes an internal model that had None as one if its inputs

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89975

### CODE REVIEW

File: test/functorch/test_aotdispatch.py [hunk 1]
```diff
@@ -364,6 +364,16 @@ def forward(self, primals_1):
     mul_1 = torch.ops.aten.mul.Tensor(mul, 3)
     return [mul, mul_1]""")
 
+    def test_input_mutation_simple_with_none(self):
+        def f(a, b):
+            return a * 3
+        inp = [torch.ones(3, 3, requires_grad=True), None]
```
Comment (aazzolini) [inline test/functorch/test_aotdispatch.py:370]: Could you add other varieties of non-tensor inputs as well? e.g. list<int> , int ?

**Review by aazzolini — APPROVED** [review]
Thanks for the fix!

**Comment by ezyang** [comment]
I'm not sure i understand this PR. Isn't the invariant we only get Tensor inputs

**Comment by aazzolini** [comment]
What is the status on this PR? There's a regression in some of our work that depends on this.

**Comment by bdhirsh** [comment]
The contents of this PR (and the feedback about making the test better) is all in https://github.com/pytorch/pytorch/pull/89532. Since that PR will take some time to land, I'll land this first.

**Pull request closed without merging**